### PR TITLE
add: LeafSignal for BehaviourTree

### PR DESCRIPTION
**Pull request by SirPigeonz**
Added new Leaf node that can emit a signal with optional custom arguments stored in an Array.

I though it would be a nice addition next to LeafCall...
 ...and I love Signals... :signal_strength: :heart:

### CODE REVIEW

File: addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd
Comment (ThePat02) [inline addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd:1]: I think this leaf should provide two features:
- Emitting its own signal
- Emitting a signal of the actor

That said, a bool `call_on_actor` paired with a string `actor_signal` would work pretty well for that. Maybe even a `call_on_leaf` bool too, so you can emit both.

File: addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@icon("res://addons/behaviour_toolkit/icons/BTLeafSignal.svg")
+class_name LeafSignal
```
Comment (ThePat02) [inline addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd:2]: I used this syntax in every script:
```gdscript
class_name LeafSignal extends BTLeaf
```
I would prefer this over a multline approach, as it makes the header smaller and more readable.

**Review by ThePat02 — CHANGES_REQUESTED** [review]
I'll take a closer look tomorrow, but it seems good!

File: addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd
Comment (SirPigeonz) [reply addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd:1]: Good idea I will add it tomorrow :)

Comment (SirPigeonz) [reply addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd:1]: I want to disscuss it more in #39 :)

Comment (SirPigeonz) [reply addons/behaviour_toolkit/behaviour_tree/leaves/leaf_signal.gd:1]: Updated the script :)

**Pull request merged**